### ISSUE TITLE
Failing to clone repository if Gitlab instance host isn't a registered domain name

### ISSUE THREAD

**Issue by blue-tachikoma** [scala-steward-org/scala-steward#2957]
I'm not sure this issue is really relevant or not because if someone uses Steward with Gitlab on-premise they have a domain name most of the time.

As I understand Steward makes request to Gitlab to retrieve repository info which includes clone url. So if my Gitlab instance hostname is `localhost` it doesn't matter what I pass to `forge-api-host` it will return real url (in this case - `localhost`) and Steward will use it to clone repository. But Steward runs inside docker so it doesn't see any repositories at this host which result in failing git clone - `fatal: unable to access 'http://localhost/steward/steward-test-1.git/': Failed to connect to localhost port 80 after 0 ms: Connection refused`.

**Comment by fthomas** [comment]
I think the easiest option here would be to not use the Docker image but to run/install Scala Steward via Coursier instead: https://github.com/scala-steward-org/scala-steward/issues/1453#issuecomment-638905164

**Comment by blue-tachikoma** [comment]
I think I need to clarify what I run and where.

I start Gitlab instance and Gitlab Runner as containers via docker compose on my local machine and have configured a pipeline job inside runner which starts Steward docker container. All of the containers are in the same docker network. So until I start Steward on host using Coursier or sbt the issue will remain. Which is not so useful because the goal was to run Steward as a scheduled pipeline job.

**Comment by fthomas** [comment]
> pipeline job inside runner which starts Steward docker container

Can't you use `cs launch` here instead of starting another Docker container?

**Comment by blue-tachikoma** [comment]
Pipeline job itself still runs inside docker container so that wouldn't have helped.

I experimented a bit and I found a way to resolve this. My problem was not correct gitlab config. I initialy setup gitlab to use `localhost` as it's hostname. And it backfired as I couldn't tell Steward to use specific hostaname for cloning repositories.

So the solution is to get rid of `localhost` at the beggining and setup docker network alias as a hostname for Gitlab instance itself. This changes clone-url from `localhost` to `myalias` and there is no need to specify it elsewhere.
